MapboxDirections/MBVisualInstruction.swift

```diff
@@ -2,127 +2,103 @@ import Foundation
 
 /**
  :nodoc:
- Encompasses all information necessary for creating a visual cue about a given `RouteStep`.
+ The contents of a banner that should be displayed as added visual guidance for a route. The banner instructions are children of the steps during which they should be displayed, but they refer to the maneuver in the following step.
  */
 @objc(MBVisualInstruction)
 open class VisualInstruction: NSObject, NSSecureCoding {
     
+    open static var supportsSecureCoding = true
+    
     /**
      :nodoc:
-     Distance in meters from the beginning of the step at which the visual instruction should be visible.
+     The plain text representation of this component.
+     
+     Use this property if `imageURLs` is an empty dictionary or if the URLs contained in that property are not yet available.
      */
-    @objc public let distanceAlongStep: CLLocationDistance
+    @objc public let text: String?
     
     /**
      :nodoc:
-     A plain text representation of `primaryTextComponents`.
+     The maneuver type for the `VisualInstruction`.
      */
-    @objc public let primaryText: String
-
-    /**
-     :nodoc:
-     Most important visual content to convey to the user about the `RouteStep`.
-     
-     This is the structured representation of `primaryText`.
-     */
-    @objc public let primaryTextComponents: [VisualInstructionComponent]
+    @objc public var maneuverType: ManeuverType
     
     /**
      :nodoc:
-     A plain text representation of `secondaryTextComponents`.
+     The modifier type for the `VisualInstruction`.
      */
-    @objc public let secondaryText: String?
+    @objc public var maneuverDirection: ManeuverDirection
     
     /**
      :nodoc:
-     Ancillary visual information about the `RouteStep`.
+     Most important visual content to convey to the user about the `RouteStep`.
      
-     This is the structured representation of `secondaryText`.
+     This is the structured representation of `text`.
      */
-    @objc public let secondaryTextComponents: [VisualInstructionComponent]?
-    
+    @objc public let textComponents: [VisualInstructionComponent]
     
     /**
      :nodoc:
-     Indicates what side of a bidirectional road the driver must be driving on. Also referred to as the rule of the road.
+     The degrees at which you will be exiting a roundabout, assuming 180 indicates going straight through the roundabout.
+     Note that this property is irrelevant unless the `maneuverType` is
      */
-    @objc public var drivingSide: DrivingSide
+    @objc public var degrees: CLLocationDegrees = 180
```

Comment:
Degrees is the unit of measurement, which is already implied by `CLLocationDegrees`; `heading` would be a better name, more consistent with the rest of the API.

Reply:
Yes, `finalHeading` sounds good: #265.